modules/core/src/main/scala/fs2/kafka/internal/WithConsumer.scala

```diff
@@ -19,37 +19,28 @@ private[kafka] object WithConsumer {
   def apply[F[_], K, V](
     settings: ConsumerSettings[F, K, V]
   )(
-    implicit F: Concurrent[F],
-    context: ContextShift[F]
+    implicit F: Async[F]
   ): Resource[F, WithConsumer[F]] = {
-    val blockerResource =
-      settings.blocker
-        .map(Resource.pure[F, Blocker])
-        .getOrElse(Blockers.consumer)
 
-    blockerResource.flatMap { blocker =>
-      Resource[F, WithConsumer[F]] {
-        settings.createConsumer
-          .flatMap(Synchronized[F].of)
-          .map { synchronizedConsumer =>
-            val withConsumer =
-              new WithConsumer[F] {
-                override def apply[A](f: KafkaByteConsumer => F[A]): F[A] =
-                  synchronizedConsumer.use { consumer =>
-                    context.blockOn(blocker) {
-                      f(consumer)
-                    }
-                  }
-              }
+    Resource[F, WithConsumer[F]] {
+      settings.createConsumer
+        .flatMap(Synchronized[F].of)
+        .map { synchronizedConsumer =>
+          val withConsumer =
+            new WithConsumer[F] {
+              override def apply[A](f: KafkaByteConsumer => F[A]): F[A] =
+                synchronizedConsumer.use { consumer =>
+                  F.defer(f(consumer))
```

Comment:
needs `blocking` here, which you can't quite use, we need to think about this

Reply:
Oh, so in CE3 we don't have a way to shift an already-suspended effect to a blocking EC?

Reply:
It seems every invocation of this involves `F.delay` - if we change this to take a thunk we can use `blocking` and simplify the call site

Reply:
> we don't have a way to shift an already-suspended effect to a blocking EC

we do (`evalOn`), but no direct access to _the_ blocking EC, for reasons